pkg/kubelet/kubelet_pods.go

```diff
@@ -387,8 +387,8 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string) (map[string]string, error) {
 	// project the services in namespace ns onto the master services
 	for i := range services {
 		service := services[i]
-		// ignore services where ClusterIP is "None" or empty
-		if !v1helper.IsServiceIPSet(service) {
+		// ignore services where ClusterIP is "None" or empty but always add the master services
+		if !v1helper.IsServiceIPSet(service) && !v1helper.IsMasterService(kl.masterServiceNamespace, masterServices, service) {
```

Comment:
I would delete the masterServices set. There is only one master service and I don't see that changing any time soon.